### PR TITLE
[Merged by Bors] - use atx index in block generator

### PR DESCRIPTION
**Pull request by dshulyak**
related: https://github.com/spacemeshos/go-spacemesh/issues/5192

### CODE REVIEW

File: blocks/utils.go [hunk 1]
```diff
@@ -238,40 +253,33 @@ func rewardInfoAndHeight(logger log.Log, cdb *datastore.CachedDB, cfg Config, pr
...
+			return 0, nil, fmt.Errorf("%w: multiple proposals with the same ATX atx %v proposal %v", errDuplicateATX, p.AtxID, p.ID())
 		}
 		events.ReportProposal(events.ProposalIncluded, p)
 	}
-	// make sure we output coinbase in a stable order.
+	atxids := maps.Keys(weights)
+	// keys in order so that everyone generates same block
 	sort.Slice(atxids, func(i, j int) bool {
 		return bytes.Compare(atxids[i].Bytes(), atxids[j].Bytes()) < 0
 	})
```
Comment (poszu) [inline blocks/utils.go:276]: How about instead of sorting it here, sort the rewards by ID in the place that creates a block?

**Review by poszu — APPROVED** [review]

File: blocks/utils.go (hunk 1 above)
Comment (dshulyak) [reply blocks/utils.go:276]: it makes sense. i am afraid to break something in this code, won't do that with this change

**Comment by dshulyak** [comment]
bors merge

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**